discovery_engine_core/core/src/engine.rs

```diff
@@ -558,12 +558,47 @@ impl Engine {
     }
 
     /// Process the feedback about the user spending some time on a document.
-    pub async fn time_spent(&mut self, time_spent: &TimeSpent) {
-        if let UserReaction::Positive | UserReaction::Neutral = time_spent.reaction {
+    pub async fn time_spent(&mut self, time_spent: TimeSpent) -> Result<(), Error> {
+        cfg_if! {
+            if #[cfg(feature = "storage")] {
+                use storage::models::TimeSpendDocumentView;
+
+                let TimeSpent {
+                    id,
+                    view_time,
+                    view_mode,
+                    // dummy values when storage is enabled
+                    smbert_embedding: _,
+                    reaction: _,
```

Comment:
:art: they could just be ignored via `..`
```suggestion
                    // dummy values when storage is enabled
                    ..
```